tests/test_core.py

```diff
@@ -422,6 +424,69 @@ def test_response(self, spec):
                 == metadata["components"]["responses"]["test_response"]
             )
 
+
+    def test_response_with_HTTPStatus_code(self, spec):
+        if int(sys.version[0]) == 2:
+            # HTTPStatus is a Python3 thing
+            return
+        else:
+            from http import HTTPStatus
+            code = HTTPStatus(200)
+            spec.path(
+                    path="/pet/{petId}",
+                    operations={"get": {"responses": {code: "test_response"}}},
+            )
+
+            assert "200" in get_paths(spec)["/pet/{petId}"]["get"]["responses"]
+
+    def test_response_with_status_code_range(self, spec):
+        status_code = "2XX"
+        def uut():
+            spec.path(
+                path="/pet/{petId}",
+                operations={"get": {"responses": {status_code: "test_response"}}},
+            )
+
+        def assertStatusCodeInResponseList():
+            return status_code in get_paths(spec)["/pet/{petId}"]["get"]["responses"]
+
+        def python_openapi_combo():
+            return (int(sys.version[0]), spec.openapi_version.major)
+
+        if python_openapi_combo() == (2, 2):
+            with warnings.catch_warnings(record=True) as w:
+                warnings.simplefilter("always")
+
+                uut()
+
+                assert len(w) == 1
+                assert issubclass(w[-1].category, UserWarning)
+                assert assertStatusCodeInResponseList() == False
+
+        elif python_openapi_combo() == (2, 3):
+            uut()
+
+            assert assertStatusCodeInResponseList() == True
+
+        elif python_openapi_combo() == (3, 2):
+            with warnings.catch_warnings(record=True) as w:
+                warnings.simplefilter("always")
+
+                uut()
+
+                assert len(w) == 1
+                assert issubclass(w[-1].category, UserWarning)
+                assert assertStatusCodeInResponseList() == False
+
+        elif python_openapi_combo() == (3, 3):
+            uut()
+
+            assert assertStatusCodeInResponseList() == True
+
+        else:
+            pytest.fail("Did not expect Python{} and OpenAPI {} combination"
+                        .format(sys.version[0], spec.openapi_version.major))
+
```

Comment:
I don't see what differs between Python versions in this test. If there is no difference, this can be greatly simplified.

Reply:
Agreed. Will fix.

Reply:
I mean I don't think you need to test Python version, do you? Couldn't you just check `spec.openapi_version.major`?

Besides, I prefer to check `spec.openapi_version.major < 3` so that we don't have to change that is it doesn't change in OAS4.

Also, and admittedly this is a matter of taste, I don't think you really need to define `uut` and `assertStatusCodeInResponseList`.

For instance, `assertStatusCodeInResponseList` is called only twice and it would be just as readable to do

```py
is spec.openapi_version.major <3:
    assert status_code not in get_paths(spec)["/pet/{petId}"]["get"]["responses"]
else:
    assert status_code not in get_paths(spec)["/pet/{petId}"]["get"]["responses"]
```

We like DRY but if you look at the other tests, this is how we generally do.

In another context, I'd write

```py
assert status_code in get_paths(spec)["/pet/{petId}"]["get"]["responses"] == spec.openapi_version.major >= 3
```

but I tend to prefer readability and use similar patterns across tests.

Notes:

- If we implement the change I suggest above, the status code is always in and the test can be factored out.

- If we kept the warning, you could also inline `uut` without repetition by always catching the warnings and assert it is emitted only on PY2.

Enough nitpicking. Thanks for your work on this and your reactivity.

Reply:
My last refactoring removed a bunch of stuff, but I should have tidied up further. The test should be a lot cleaner now.

And no problem, thank you for the quick feedback and review comments :).